packages/core/src/services/evaluations/connect.ts

```diff
@@ -1,77 +1,101 @@
 import {
-  DocumentVersion,
-  EvaluationDto,
-  EvaluationMetadataType,
+  ConnectedEvaluation,
   EvaluationMode,
-  EvaluationTemplateWithCategory,
+  SafeWorkspace,
+  Workspace,
 } from '../../browser'
 import { database } from '../../client'
-import { findWorkspaceFromDocument } from '../../data-access'
-import { ErrorResult, NotFoundError, Result, Transaction } from '../../lib'
+import {
+  NotFoundError,
+  PromisedResult,
+  Result,
+  Transaction,
+  TypedResult,
+} from '../../lib'
+import {
+  DocumentVersionsRepository,
+  EvaluationsRepository,
+} from '../../repositories'
 import { connectedEvaluations } from '../../schema'
-import { createEvaluation } from './create'
+import { importLlmAsJudgeEvaluation } from './create'
 
 export function connectEvaluations(
   {
-    document,
-    templates,
-    evaluations: evaluationToImport,
+    workspace,
+    documentUuid,
+    evaluationUuids,
+    templateIds,
     evaluationMode = EvaluationMode.Batch,
   }: {
-    document: DocumentVersion
-    templates: EvaluationTemplateWithCategory[]
-    evaluations: EvaluationDto[]
+    workspace: Workspace | SafeWorkspace
+    documentUuid: string
+    evaluationUuids?: string[]
+    templateIds?: number[]
     evaluationMode?: EvaluationMode
   },
   db = database,
-) {
-  return Transaction.call(async (tx) => {
-    const workspace = await findWorkspaceFromDocument(document, tx)
-    if (!workspace) {
-      return Result.error(new NotFoundError('Workspace not found'))
-    }
-
-    // TODO: Creating an evaluation is kind of a pita because of the
-    // polymorphic relation with metadata so we use the creation service which
-    // causes N db operations (not ideal). Implement a bulkCreate of
-    // evaluations service.
-    const results = await Promise.all(
-      templates.map((template) =>
-        createEvaluation(
-          {
-            workspace,
-            name: template.name,
-            description: template.description,
-            type: EvaluationMetadataType.LlmAsJudge,
-            metadata: {
-              prompt: template.prompt,
-            },
-          },
-          tx,
-        ),
-      ),
-    )
+): PromisedResult<ConnectedEvaluation[], Error> {
+  return Transaction.call(
+    async (tx): PromisedResult<ConnectedEvaluation[], Error> => {
+      const documentVersionsScope = new DocumentVersionsRepository(
```

Comment:
I need to use a transaction, because it creates new evaluations from the templates before creating the connections! If anything fails in between those two actions, I'd like to undo the changes.